libraries/lib-import-export/PlainExportOptionsEditor.cpp

```diff
@@ -0,0 +1,106 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  PlainExportOptionsEditor.cpp
+
+  Vitaly Sverchinsky
+
+**********************************************************************/
+
+#include "PlainExportOptionsEditor.h"
+#include <wx/config.h>
+
+PlainExportOptionsEditor::PlainExportOptionsEditor(std::initializer_list<OptionDesc> options)
+{
+   mOptions.reserve(options.size());
+   mValues.reserve(options.size());
+   for(auto& desc : options)
+   {
+      mValues[desc.option.id] = desc.option.defaultValue;
+      mOptions.emplace_back(desc.option);
+      mConfigKeys.push_back(desc.configKey);
+   }
+}
+
+int PlainExportOptionsEditor::GetOptionsCount() const
+{
+   return static_cast<int>(mOptions.size());
+}
+
+bool PlainExportOptionsEditor::GetOption(int index, ExportOption& option) const
+{
+   if(index >= 0 && index < static_cast<int>(mOptions.size()))
+   {
+      option = mOptions[index];
+      return true;
+   }
+   return false;
+}
+
+bool PlainExportOptionsEditor::GetValue(int id, ExportValue& value) const
+{
+   const auto it = mValues.find(id);
+   if(it != mValues.end())
+   {
+      value = it->second;
+      return true;
+   }
+   return false;
+}
+
+bool PlainExportOptionsEditor::SetValue(int id, const ExportValue& value)
+{
+   const auto it = mValues.find(id);
+   if(it != mValues.end() && it->second.index() == value.index())
+   {
+      it->second = value;
+      return true;
+   }
+   return false;
+}
+
+void PlainExportOptionsEditor::Load(const wxConfigBase& config)
+{
+   auto index = 0;
+   for(const auto& option : mOptions)
+   {
+      auto it = mValues.find(option.id);
+      assert(it != mValues.end());
+      if(auto val = std::get_if<bool>(&it->second))
+         config.Read(mConfigKeys[index], val);
+      else if(auto val = std::get_if<int>(&it->second))
+         config.Read(mConfigKeys[index], val);
+      else if(auto val = std::get_if<double>(&it->second))
+         config.Read(mConfigKeys[index], val);
+      else if(auto val = std::get_if<std::string>(&it->second))
+      {
+         wxString wxstr;
+         if(config.Read(mConfigKeys[index], &wxstr))
+            *val = wxstr.ToStdString();
+      }
```

Comment:
I could use my new utility like this, avoiding repetition, and requiring no rewriting in case there is ever a need
to add more variant alternatives that the generic lambda handles:

```
Visit(Variant::MakeVariant(
   [](auto &val) { config.Read(mConfigKeys[index], &val); },
   [](wxString &val) {
      wxString wxStr;
      if(config.Read(mConfigKeys[index], &wxstr))
         *val = wxstr.ToStdString();
   }
), it->second);
```

Similarly in other places where you use `get_if`